python_modules/dagster-graphql/dagster_graphql/implementation/fetch_auto_materialize_asset_evaluations.py

```diff
@@ -1,39 +1,60 @@
-from typing import Optional
+from typing import TYPE_CHECKING, Optional
 
-from dagster import AssetKey, DagsterInstance
+from dagster import AssetKey
+from dagster._core.host_representation.external_data import ExternalAssetNode
 
+from dagster_graphql.implementation.fetch_assets import asset_node_iter
 from dagster_graphql.schema.auto_materialize_asset_evaluations import (
     GrapheneAutoMaterializeAssetEvaluationNeedsMigrationError,
     GrapheneAutoMaterializeAssetEvaluationRecord,
     GrapheneAutoMaterializeAssetEvaluationRecords,
 )
 from dagster_graphql.schema.inputs import GrapheneAssetKeyInput
 
+if TYPE_CHECKING:
+    from ..schema.util import ResolveInfo
+
 
 def fetch_auto_materialize_asset_evaluations(
-    instance: DagsterInstance,
-    asset_key: GrapheneAssetKeyInput,
+    graphene_info: "ResolveInfo",
+    graphene_asset_key: GrapheneAssetKeyInput,
     limit: int,
     cursor: Optional[str],
 ):
     """Fetch asset policy evaluations from storage."""
-    if instance.schedule_storage is None:
+    if graphene_info.context.instance.schedule_storage is None:
         return GrapheneAutoMaterializeAssetEvaluationNeedsMigrationError(
             message="Instance does not have schedule storage configured, cannot fetch evaluations."
         )
-    if not instance.schedule_storage.supports_auto_materialize_asset_evaluations:
+    if (
+        not graphene_info.context.instance.schedule_storage.supports_auto_materialize_asset_evaluations
+    ):
         return GrapheneAutoMaterializeAssetEvaluationNeedsMigrationError(
             message=(
                 "Auto materialize evaluations are not getting logged. Run `dagster instance"
                 " migrate` to enable."
             )
         )
 
+    asset_key = AssetKey.from_graphql_input(graphene_asset_key)
+
+    asset_node: Optional[ExternalAssetNode] = None
+    for _, _, external_asset_node in asset_node_iter(graphene_info):
```

Comment:
I think we should use `get_asset_nodes_by_asset_key` instead.

`asset_node_iter` iterates through all code locations, repositories, and asset in order, which means you might get a source asset with the same key rather than the actual asset node. This is probably fine with the current implementation, but may lead to unexpected behaviors if we do more things with the external asset node in the future